src/game/Grids/GridNotifiersImpl.h

```diff
@@ -284,6 +284,8 @@ inline void MaNGOS::DynamicObjectUpdater::VisitHelper(Unit* target)
         if (!holder->GetAuraByEffectIndex(eff_index))
         {
             PersistentAreaAura* Aur = new PersistentAreaAura(spellInfo, eff_index, &i_dynobject.GetDamage(), &i_dynobject.GetBasePoints(), holder, target, caster);
+            if (!i_dynobject.OnAreaAuraCheckTarget(Aur, target))
```

Comment:
Pass dyngo as param and do the check way higher before any aura creation. Also, you just wrote a memleak :D

Reply:
Yeah, forgot that pointers don't go out of scope :D been rusty